File: tests/test_code/php/chains/chains.php
<?php
function a() {
    echo "A";
    b();
}


function b() {
    echo "B";
    a();
}


class Cls {
    function __construct() {
        echo "init";
        return $this;
    }


    function a() {
        echo "a";
        return $this;
    }

    function b() {
        echo "b";
        return $this;
    }
}


function c() {
    $obj = new Cls();
    $obj->a()->b()->a();
}

a();
b();
c();
?>

File: tests/test_code/rb/chains/chains.rb
def a
    puts "A"
    b
end

def b
    puts "B"
    a
end

class Cls
    def initialize
        puts "init"
    end

    def a
        puts "a"
        self
    end

    def b
        puts "b"
        return self
    end
end

def c
    obj = Cls.new()
    obj.a.b.a.c
end

a
b
c
